src/app/providers/examiner-records/examiner-records.ts

```diff
@@ -0,0 +1,239 @@
+import { get } from 'lodash-es';
+import { QuestionResult } from '@dvsa/mes-test-schema/categories/common';
+import { DateRange } from '@shared/helpers/date-time';
+import { ChartType } from 'ng-apexcharts';
+import { Router } from '@angular/router';
+import { LoadingProvider } from '@providers/loader/loader';
+import moment from 'moment';
+
+export interface ColourScheme {
+  name: ColourEnum,
+  bar: string[],
+  pie: string[],
+  emergencyStop?: string[],
+  average: string
+}
+
+export const enum ColourEnum {
+  DEFAULT = 'Default',
+  GREYSCALE = 'Greyscale',
+}
+
+export type ExaminerRecordsRange = DateRange;
+
+export interface SelectableDateRange {
+  display: string;
+  val: ExaminerRecordsRange;
+}
+export type DESChartTypes = Extract<ChartType, 'bar' | 'pie'>;
+@Injectable()
+export class ExaminerRecordsProvider {
+
+  public colours: {
+    default: ColourScheme,
+    greyscale: ColourScheme,
+  } = {
+    default: {
+      name: ColourEnum.DEFAULT,
+      pie: [
+        '#008FFB',
+        '#ED6926',
+        '#FF526F',
+        '#007C42',
+        '#a05195',
+      ],
+      bar: ['#008FFB'],
+      emergencyStop: [
+        '#ED6926',
+        '#777777'
+      ],
+      average: '#000000',
+    },
+    greyscale: {
+      name: ColourEnum.GREYSCALE,
+      pie: [
+        '#474747',
+        '#6E6E6E',
+        '#222222',
+        '#606060',
+        '#949494',
+      ],
+      bar: ['#777777'],
+      average: '#000000',
+    },
+  };
+
+  public localFilterOptions: SelectableDateRange[] = [
+    {
+      display: 'Today',
+      val: 'today',
+    },
+    {
+      display: 'Last 7 days',
+      val: 'week',
+    },
+    {
+      display: 'Last 14 days',
+      val: 'fortnight',
+    },
+  ];
+  public onlineFilterOptions: SelectableDateRange[] = [
+    {
+      display: 'Last 90 days',
+      val: '90 days',
+    },
+    {
+      display: 'Last 1 year',
+      val: '1 year',
+    },
+    {
+      display: 'Last 18 months',
+      val: '18 months',
+    },
+  ];
+
+  currentlyLoading: boolean = false;
+
+
+  constructor(
+    public searchProvider: SearchProvider,
+    public compressionProvider: CompressionProvider,
+    public store$: Store<StoreModel>,
+    public router: Router,
+    public loadingProvider: LoadingProvider,
+  ) {
+  }
+
+  /**
+   * handler for loading spinner while pulling backend data
+   */
+  handleLoadingUI = async (isLoading: boolean) => {
+    if ((isLoading && !this.currentlyLoading) || (!isLoading && this.currentlyLoading)) {
+      this.currentlyLoading = isLoading;
+      await this.loadingProvider.handleUILoading(isLoading, {
+        id: 'examinerRecord_loading_spinner',
+        spinner: 'circles',
+        backdropDismiss: false,
+        translucent: false,
+        message: 'Loading...'
+      });
+
+    }
+    return null;
+  };
+
+  /**
+  Get the date from a set date range ago in order to display on screen
+   */
+  getRangeDate(range: DateRange): moment.Moment {
```

Comment:
can we remove the breaks.

getRangeDate(range: DateRange): moment.Moment {
  const today = new Date();
  switch (range) {
    case 'today':
      return moment(today);
    case 'week':
      return moment(today).subtract(1, 'week');
    case 'fortnight':
      return moment(today).subtract(2, 'weeks');
    case '90 days':
      return moment(today).subtract(90, 'days');
    case '1 year':
      return moment(today).subtract(1, 'year');
    case '18 months':
      return moment(today).subtract(18, 'months');
    default:
      return null;
  }
}

Reply:
getRangeDate(range: DateRange): moment.Moment {
  const today = new Date();
  switch (range) {
    case DateRange.TODAY:
      return moment(today);
    case DateRange.WEEK:
      return moment(today).subtract(1, 'week');
    case DateRange.FORTNIGHT:
      return moment(today).subtract(2, 'weeks');
    case DateRange.NINETY_DAYS:
      return moment(today).subtract(90, 'days');
    case DateRange.ONE_YEAR:
      return moment(today).subtract(1, 'year');
    case DateRange.EIGHTEEN_MONTHS:
      return moment(today).subtract(18, 'months');
    default:
      return null;
  }
}